urdfenvs/urdfCommon/generic_robot.py

```diff
@@ -65,18 +55,34 @@ def read_limits(self) -> None:
     def set_acceleration_limits(self):
         pass
 
-    def get_indexed_joint_info(self) -> dict:
-        """Get indexed joint info.
+    def get_indexed_joint_info(self) -> None:
+        """Automated extraction of joint ids
 
-        This function can be used for debugging and finding
-        the correct joint indices for self.setJointIndices.
+        Extract joint ids by the joint names.
 
         """
-        indexed_joint_info = {}
-        for i in range(p.getNumJoints(self._robot)):
+        if not hasattr(self, '_joint_names'):
+            return
+        robot = URDF.load(self._urdf_file) 
+        self._urdf_joints = [] 
+        for i, joint in enumerate(robot.joints): 
+            if joint.name in self._joint_names: 
+                self._urdf_joints.append(i) 
+        self._robot_joints = []
+        self._castor_joints = []
+        num_joints = p.getNumJoints(self._robot)
+        for name in self._joint_names:
+            for i in range(num_joints):
+                joint_info = p.getJointInfo(self._robot, i)
+                joint_name = joint_info[1].decode("UTF-8")
+                if joint_name == name:
+                    self._robot_joints.append(i)
```

Comment:
This double loop seems to be the only solution for now.
The complexity does not matter for the moment.